config.py

```diff
@@ -15,8 +15,16 @@
         # Bot settings
         Validator("log_level", is_type_of=str),
         Validator("bot_token", must_exist=True, is_type_of=str),
+        # Redis settings
+        Validator("redis_host", must_exist=True, is_type_of=str),
+        Validator("redis_port", must_exist=True, is_type_of=int),
+        Validator("redis_username", is_type_of=str),
+        Validator("redis_password", is_type_of=str),
+        Validator("redis_key_const", is_type_of=str),
         # Roulette settings
+        Validator("roulette_unmute_rate",  is_type_of=int),
         Validator("roulette_channels", must_exist=True, is_type_of=list, len_min=1),
+        Validator("roulette_timeout_roles", must_exist=True, is_type_of=list, len_min=1),
```

Comment:
note: This will be a breaking change, and it's probably better to ensure the bot still supports a no-role version (pure timeout only). Let's remove `must_exist=True` (and `len_min`) here - the actual role application code can just iterate over an empty list and no-op.

Reply:
While this works fine, it disables some logging messages after the timeout passed. Shouldn't be a problem with large data sets, though I will mark it as a todo.